src/ensembl/src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields.tsx

```diff
@@ -1,115 +1,148 @@
 import React, { useState, useEffect } from 'react';
 import Checkbox from 'src/shared/components/checkbox/Checkbox';
-import styles from './CheckboxWithTextfields.scss';
 
 import ImageButton from 'src/shared/components/image-button/ImageButton';
-import { ReactComponent as AddIcon } from 'static/img/browser/zoom-in.svg';
-import PasteOrUpload from '../paste-or-upload/PasteOrUpload';
+import { ReactComponent as RemoveIcon } from 'static/img/shared/clear.svg';
+
+import Textarea from 'src/shared/components/textarea/Textarea';
+import Upload, { ReadFile } from 'src/shared/components/upload/Upload';
+import { nextUuid } from 'src/shared/helpers/uuid';
```

Comment:
I missed this previously, but we do we call this helper `uuid`? uuid is a universally unique identifier; it has several possible formats, but just being a regularly incremented number is not one of them.

Reply:
Do you have any suggestions for what to call it as?

Reply:
just an `id`? `generateSimpleId` instead of `nextUuid`?

Reply:
How about `generateUniqueId` ? I feel like the main thing is missing if we say  `generateSimpleId`

Reply:
Ids should be unique by definition 😄

Reply:
That's true but would it not be a little bit more helpful if we use `Unique` instead of `Simple` rather than having to go by definition?

Reply:
The word "unique" is redundant, and misleading. By calling a function `generateUniqueId`, we imply that there may also be non-unique id's that some other function may generate.

I am not particularly attached to the word "simple". `generateId`? `generateNumericId`? `generateId({ strategy: 'increment-number' })`? The `uuid` [package](https://www.npmjs.com/package/uuid) on npm refers to `uuid/v1`, `uuid/v3`, etc. (although each of these versions have technical definitions); so we could take inspiration from them and do something like `generateId({ version: 1 })`; or `getIdGenerator({ version: 1 })`, which returns a function that generates ids by incrementing numbers (if we expect any other id generation strategies to warrant introducing versions).

Reply:
I guess we can go for `generateId` now and if we need to introduce versioning, it can be done in a separate PR.